kong/runloop/handler.lua

```diff
@@ -405,44 +422,59 @@ do
   end
 
 
-  local function get_version(name)
-    if not kong.cache then
-      return "init"
+  local function log_with_worker_id(level, message)
+    if WORKER_ID then
+      log(level, message, " (worker id: ", WORKER_ID, ")")
+    else
+      log(level, message)
     end
+  end
 
-    local version, err = kong.cache:get(name .. ":version", TTL_ZERO, utils.uuid)
-    if err then
-      return nil, "could not ensure " .. name .. " is up to date: " .. err
-    end
 
-    return version
-  end
+  local function safe_rebuild(name, callback, version)
+    local max_attempts = 6
```

Comment:
I think in original code I tried 5 times, but this is just a number thrown out of hat.

Reply:
I replaced this with a parameter. The parameter is set to 5 on timer-based rebuilds and to 1 on request-based rebuild. Request-based rebuilds use stale result if a semaphore fails, or a version fails.